docs/rfc/active/000-internationalization.md

```diff
@@ -0,0 +1,165 @@
+> 
+
+> In any case, if the translation doesn't exist in requested language, then it give the default (ie. English) translation.
+> If it doesn't exist in the default language, then it return the given translation key.
+> So the worste case scenario is an ugly translation, but no blocking bugs.
+>
+
+> By default, it can be configured to use the browser language
+> (with a fallback to the default language if browser language is not translated).
+>
+
+> In practice, I think the lightest way for the Datahub team would be to just implement the english language. 
+> Leaving any other language for others devs. Since we should be able to dynamically check how many language there is and which one are they,
+> we could add a "Change Language" field in the Header Menu opening a modal where you can select your desired language.
+> And obviously, we hide it if there is only one language selectable.
+
+
+### Corner-case
+
+#### Front end test
+> There are tests in the frontend looking for the presence of text. I have deactivated them.
+> I am not sure how react-i18next will render in a test environment.
+> 
+
+#### Back end text
+> Some text come from the back end.
+> - The back-end exceptions : we can handle then by adding a translate key to exceptions, and the front will
+    > have to apply i18next on the the translate key to display it.
+> - The whole Analytics page : It is one of the majors problem... All text of the Analytics page is built,
+    > almost word by word, by the gms... I am not sure what is the best way here... Maybe moving theses constructs
+    > from the back to the front...
+
+#### Text size
+> Depending on the language, some text length may vary. This could be a problem on buttons or narrow panel for exemple
+> 
+
+#### Other libraries
+> When using other libraries, they may not have an i18n interface. For example the yaml editor in *YamlEditor.tsx*.
+> 
+
+## How we teach this
+> Since I am thinking about adding only the architecture for the i18n, it will be usefull only for developers.
+> But any developer who want to add a new language can follow some very brief instructions, a single page for internationalization 
+> should be enough. The only thing to do is to copy an English *translation.json* and write it in desired language.
+>
+
+## Drawbacks
+> The unique but wearisome drawbacks is that the developers won't be able to write direct text in their 
+> component / exceptions. They'll have to report, and distribute correctly, their text in a *translation.json* file.
+> 
+
+## Alternatives
+What other designs have been considered?
+> None
+>
+
+What is the impact of not doing this?
+> None
+>
+
+## Rollout / Adoption Strategy
+> Once it is done, there won't be any modifications to do for people whom use Datahub as it is.
+> For anyone who have made a fork, it could have an impact if they added type, subtype or any generic word,
```

Comment:
Well as long as its optimistic translation custom types, subtypes may not be a problem (they just won't undergo translation)